### PR TITLE
feat(Examples): Firmware update through HCI for MAX32655

### PR DESCRIPTION
**Pull request by JC-282-AD**
### Description

Write a firmware update example: update the firmware through HCI
Add HCI command to be used during the update: erase, update, sysreset

### CODE REVIEW

File: Libraries/Cordio/controller/sources/ble/lhci/lhci_cmd_vs.c [hunk 1]
```diff
@@ -83,10 +92,76 @@ bool_t lhciCommonVsStdDecodeCmdPkt(LhciHdr_t *pHdr, uint8_t *pBuf)
...
+        uint32_t addr;
+        uint32_t size;
+
+
+        BSTREAM_TO_UINT32(addr, pBuf);
+        address = (uint32_t*) addr;
+        uint8_t * addressErase = (uint8_t*) addr;
+
+        BSTREAM_TO_UINT32(size, pBuf);
+        volatile uint32_t address32 = (uint32_t)addressErase;
```
Comment (EricB-ADI) [inline Libraries/Cordio/controller/sources/ble/lhci/lhci_cmd_vs.c:123]: What is addresss32 for? I only see it set

File: Libraries/Cordio/controller/sources/ble/lhci/lhci_cmd_vs.c [hunk 2]
```diff
@@ -64,6 +66,13 @@
 /**************************************************************************************************
   Functions
 **************************************************************************************************/
+//this is the callback function to the Msg
+void device_reset(wsfEventMask_t event, wsfMsgHdr_t *pMsg)
+{
+    MXC_Delay(3000);
```
Comment (EricB-ADI) [inline Libraries/Cordio/controller/sources/ble/lhci/lhci_cmd_vs.c:72]: Do you need the delay?

Comment (JC-282-AD) [reply Libraries/Cordio/controller/sources/ble/lhci/lhci_cmd_vs.c:72]: > Do you need the delay?

I think yes, without the delay, the reset will be completed before HCI sends the status code back.

File: Libraries/Cordio/controller/sources/ble/lhci/lhci_cmd_vs.c (hunk 1 above)
Comment (JC-282-AD) [reply Libraries/Cordio/controller/sources/ble/lhci/lhci_cmd_vs.c]: > What is addresss32 for? I only see it set

Sorry, this becomes unused after several changes. Deleted.

File: Libraries/Cordio/controller/sources/ble/lhci/lhci_cmd_vs.c (hunk 2 above)
Comment (EricB-ADI) [reply Libraries/Cordio/controller/sources/ble/lhci/lhci_cmd_vs.c:72]: That delay is blocking. meaning nothing is going on when you run it.

Comment (EricB-ADI) [reply Libraries/Cordio/controller/sources/ble/lhci/lhci_cmd_vs.c:74]: Instead of doing an event with a delay just use a one 
shot timer

**Review by EricB-ADI — APPROVED** [review]

**Pull request merged**